dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "DCPS/DdsDcps_pch.h"
+#include "TypeLookupService.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+TypeLookupService::TypeLookupService()
+{
+  to_empty_.minimal.kind = TK_NONE;
+  to_empty_.complete.kind = TK_NONE;
+  rpc_sequence_number_ = 0;
+}
+
+
+TypeLookupService::~TypeLookupService()
+{
+}
+
+
+void TypeLookupService::get_type_objects(const TypeIdentifierSeq& type_ids,
+                                         TypeIdentifierTypeObjectPairSeq& types)
+{
+  for (CORBA::ULong i = 0; i < type_ids.length(); ++i) {
+    TypeObjectMap::iterator it_object = type_object_map_.find(type_ids[i]);
+    if (it_object != type_object_map_.end()) {
+      TypeIdentifierTypeObjectPair new_to_pair(it_object->first, it_object->second);
```

Comment:
const it, or just combine it with the next line (direct constructor call to make a temporary object).